filebeat/fileset/pipelines.go

```diff
@@ -178,6 +183,42 @@ func setECSProcessors(esVersion common.Version, pipelineID string, content map[s
 	return nil
 }
 
+func checkProcessorMinVersion(esVersion common.Version, pipelineID string, content map[string]interface{}) error {
+	ecsVersion := common.MustNewVersion("7.0.0")
```

Comment:
If the intention is to be able to use this in a bit more generic way, maybe doing something like:

```
func setECSProcessors(esVersion common.Version, pipelineID string, content map[string]interface{}) error {
	versionChecks := map[string]struct{
		minVersion common.Version
		fn func(esVersion common.Version, pipelineID string, content map[string]interface{}) error
	}{
		"uri_parts": {
			minVersion: common.MustNewVersion("7.12"),
		},
	}

	// ...

	for i, p := range processors {
		processor, ok := p.(map[string]interface{})
		if !ok {
			continue
		}
		
		for pname, check := range versionChecks {
			p, found := processor[pname]
			if !found {
				newProcessors = append(newProcessors, p)
				continue
			}

			if _, ok := processor[pname].(map[string]interface{}); ok {
				if !esVersion.LessThan(check.minVersion) {
					continue
				}
				
				logp.Debugf("processors", "Current version of Elasticsearch: %v does not support the %s processor, minimum version is: %v and newer)", esVersion, pname, minURIPartsVersion)
				if check.fn != nil {
					if err := check.fn(esVersion, pipelineID, content); err != nil {
						return err
					}
				}
			}
		}

		newProcessors = append(newProcessors, processors[i])

	}
	content["processors"] = newProcessors
}
```

(wrote that in the browser so is more than sure very wrong, but just to tell the idea).

This way some of the other checks that are done could be centralized here. Does it make sense?

Reply:
@marc-gr Should be maybe have a similar setup for options/arguments as well then, and not just whole processors?